terraform/provider/provider.go

```diff
@@ -37,27 +42,39 @@ func Provider() *schema.Provider {
 			},
 		},
 		ResourcesMap: map[string]*schema.Resource{
-			"teleport_user": resourceTeleportUser(),
 			"teleport_role": resourceTeleportRole(),
+			"teleport_user": resourceTeleportUser(),
 		},
-		ConfigureFunc: providerConfigure,
+		ConfigureContextFunc: configureContext,
 	}
 }
 
-// providerConfigure accepts provider settings schema,
+// configure accepts provider settings schema,
 // initializes a Teleport's `auth.Client` and returns it.
-func providerConfigure(d *schema.ResourceData) (interface{}, error) {
-	certPath := d.Get("cert_path").(string)
-	keyPath := d.Get("key_path").(string)
-	rootCAsPath := d.Get("root_ca_path").(string)
+func configureContext(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
+	certPath, err := getPath(d, "cert_path")
+	if err != nil {
+		return nil, diag.FromErr(trace.Wrap(err))
+	}
+
+	keyPath, err := getPath(d, "key_path")
+	if err != nil {
+		return nil, diag.FromErr(trace.Wrap(err))
+	}
 
-	// TODO: Verify Addr format is [protocol://]host:port
-	addr := d.Get("addr").(string)
+	rootCAsPath, err := getPath(d, "root_ca_path")
+	if err != nil {
+		return nil, diag.FromErr(trace.Wrap(err))
+	}
+
+	addr, err := getAddr(d, "addr")
+	if err != nil {
+		return nil, diag.FromErr(trace.Wrap(err))
+	}
 
 	tlsConfig, err := loadTLSConfig(certPath, keyPath, rootCAsPath)
```

Comment:
If/when you update to the latest api client, you'll be able to use `client.LoadKeyPair(certPath, keyPath, rootCAsPath)`, or `client.LoadTLS(tlsConfig)` if you want to maintain custom errors and such.

Reply:
Done, could you please check if I implemented it correctly? (provider.go#75). Am I correct that I should not be handling errors at this line (when I call the loader), and handling error returned by the client.New() call is sufficient?

Reply:
Yes, that's all correct.